File: Python/784. Letter Case Permutation.py
class Solution:
    def letterCasePermutation(self, S):
        """
        :type S: str
        :rtype: List[str]
        """
        ans = ['']
        for ch in S:
            if ch.isalpha():
                ans = [i + j for i in ans for j in [ch.upper(), ch.lower()]]
            else:
                ans = [i + ch for i in ans]
        return ans

File: C++/784. Letter Case Permutation.cpp
class Solution {
public:
    void dfs(string s,int i,vector<string> &ans)
    {
        if(i==s.length())
        {
            ans.push_back(s);
            return ;
        }
        if(isalpha(s[i]))
        {
            if(s[i]>='a')
                s[i] = s[i]-'a'+'A';
            else
                s[i] = s[i]-'A'+'a';
            dfs(s,i+1,ans);
            if(s[i]>='a')
                s[i] = s[i]-'a'+'A';
            else
                s[i] = s[i]-'A'+'a';
        }
        dfs(s,i+1,ans);
    }
    vector<string> letterCasePermutation(string S) {
        vector<string> ans;
        dfs(S,0,ans);
        return ans;
    }
};